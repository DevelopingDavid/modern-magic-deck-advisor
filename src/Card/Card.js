import React, { Component } from 'react';
import '../styles/main.scss';

class Card extends Component {
  setPopUpData = () => {
    this.props.displayPopUp(this.props.card);
  };

  addToWishlist = () => {
    this.props.addToWishlist(this.props.card);
  };

  render() {
    return (
      <article className="card--container">
        {this.props.cardAreaView === 'myCardList' && (
          <div>
            <div className="card--header-div">
              <h2>{this.props.card.cardName}</h2>
            </div>
            <img
              className="card--image"
              src={this.props.card.imageSource}
              alt={this.props.card.cardName}
              onClick={this.setPopUpData}
            />
            <div className="card--counter-container">
              <p className="card--counter">
                {this.props.card.cardCount} in My Cards
              </p>
            </div>
          </div>
        )}
        {this.props.cardAreaView === 'expandedDeck' && (
          <div>
            <div className="card--header-div">
              <h2>{this.props.card.cardName}</h2>
            </div>
            <p>${(this.props.card.price).toFixed(2)}</p>
            <img
              className="card--image"
              src={this.props.card.imageSource}
              alt={this.props.card.cardName}
              onClick={this.setPopUpData}
            />
            <div className="card--counter-container">
              <p>
                {`You have ${this.props.card.userCount}
                   of ${this.props.card.requiredCount}`}
              </p>
              <button onClick={this.addToWishlist}>Add to Wishlist</button>
            </div>
          </div>
        )}
        {this.props.cardAreaView === 'wishList' && (
          <div>
            <div className="card--header-div">
              <h2>{this.props.card.cardName}</h2>
            </div>
            <img
              className="card--image"
              src={this.props.card.imageSource}
              alt={this.props.card.cardName}
              onClick={this.setPopUpData}
            />
            <div>
              <p className="card--wishlist-price">
                Total Cost of {this.props.card.wishListCount} - $
                {(
                  this.props.card.price * this.props.card.wishListCount
                ).toFixed(2)}
              </p>
            </div>
          </div>
        )}
      </article>
    );
  }
}

export default Card;
