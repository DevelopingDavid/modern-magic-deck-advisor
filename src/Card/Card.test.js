import React from 'react';
import Card from './Card';
import { shallow } from 'enzyme';

const displayPopUpMock = jest.fn();
const addToWishlistMock = jest.fn();
const card = { cardName: 'Black Lotus' };
const getPriceStringMock = jest.fn()

describe('Card', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <Card
      card={card} 
      key={card.cardName}
      cardAreaView={'expandedDeck'}
      displayPopUp={displayPopUpMock}
      addToWishlist={addToWishlistMock}
      getPriceString={getPriceStringMock}/>
    );
  });

  it('should match the snapshot when cardAreaView is myCardList', () => {
    let wrapper = shallow(
      <Card
        card={card} 
        key={card.cardName}
        cardAreaView={'myCardList'}
        displayPopUp={displayPopUpMock}
        getPriceString={getPriceStringMock}/>
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should match the snapshot when cardAreaView is wishList', () => {
    let wrapper = shallow(
      <Card
        card={card}
        key={card.cardName}
        cardAreaView={'wishList'}
        displayPopUp={displayPopUpMock}
        getPriceString={getPriceStringMock}/>
    );
    expect(wrapper).toMatchSnapshot();
  })
  
  it('should match the snapshot when cardAreaView is expandedDeck', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should call props.displayPopUp when the image is clicked', () => {
    wrapper.find('.card--image').simulate('click');
    expect(displayPopUpMock).toBeCalled();
  });
  
  it('should call props.addToWishlist when the button is clicked', () => {
    wrapper.find('button').simulate('click');
    expect(addToWishlistMock).toBeCalled();
  });
});