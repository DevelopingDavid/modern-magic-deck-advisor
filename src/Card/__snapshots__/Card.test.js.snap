// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should match the snapshot when cardAreaView is expandedDeck 1`] = `
<article
  className="card--container"
>
  <div>
    <img
      alt="Black Lotus"
      className="card--image"
      onClick={[Function]}
    />
    <div
      className="counter--container"
    >
      <p />
      <p>
        You have undefined
                   of undefined
      </p>
      <button
        onClick={[Function]}
      >
        Add to Wishlist
      </button>
    </div>
  </div>
</article>
`;

exports[`Card should match the snapshot when cardAreaView is myCardList 1`] = `
<article
  className="card--container"
>
  <div>
    <img
      alt="Black Lotus"
      className="card--image"
      onClick={[Function]}
    />
    <div
      className="card--counter-container"
    >
      <h1
        className="card--counter"
      >
        X
      </h1>
    </div>
  </div>
</article>
`;
